### PR TITLE
Submit clears all fields

### PR DESCRIPTION
**Pull request by Mo-sissoko**
### ℹ️ Issue

Closes  https://trello.com/c/7LAH8dqA/11-successful-submission-of-form-should-clear-all-fields

### 📝 Description

Briefly list the changes made to the code:
- 
- Change NumberInput to use Input instead of NumberInputField
- Update FormValues to be of type Record<string, unknown> to account for nested fieldNames, e.g. medicalForms.childsName

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

Provide screenshots of any new components, styling changes, or pages. 

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

### CODE REVIEW

File: apps/frontend/src/components/form/GrantForm.tsx [hunk 1]
```diff
@@ -5,6 +5,27 @@ import React from 'react';
 import FormField from './FormField';
 import { StatesDropdownValues } from '../../enums/DropdownValues';
 
+export const grantFormInitialValues = {
```
Comment (nate-kirschner) [inline apps/frontend/src/components/form/GrantForm.tsx]: Are you able to generate this from the schema that we already have defined? Seems like theres a getDefault function on Yup schemas, but I haven't looked too much into it.
We want to limit the number of places we would have to make changes if the schema changes.

File: apps/frontend/src/components/form/MedicalForm.tsx [hunk 2]
```diff
@@ -7,6 +7,29 @@ import {
   StatesDropdownValues,
 } from '../../enums/DropdownValues';
 
+export const medicalFormInitialValues = {
```
Comment (nate-kirschner) [inline apps/frontend/src/components/form/MedicalForm.tsx]: Same comment as above

Comment (Mo-sissoko) [reply apps/frontend/src/components/form/MedicalForm.tsx]: Done in 7e5aa44.

File: apps/frontend/src/components/form/GrantForm.tsx (hunk 1 above)
Comment (Mo-sissoko) [reply apps/frontend/src/components/form/GrantForm.tsx]: Done in 7e5aa44.

**Comment by nate-kirschner** [comment]
The changes look good but I'm not able to actually submit a form on your branch. Nothing happens when I click the submit button and no network request is sent. I checked out main and was able to submit one on that so maybe one of the changes here is causing the bug?

**Review by nate-kirschner — APPROVED** [review]

**Pull request merged**